Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2465,6 +2473,14 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
             }
         }
 
+        if (node->op() == CompareStrictEq && right && right.isNumber()) {
+            double rightValue = right.asNumber();
+            if (std::isnan(rightValue)) {
+                setConstant(node, jsBoolean(false));
+                break;
+            }
+        }
```

Comment:
Ditto.

Reply:
`CompareStrictEq` doesn't call `valueOf` so I think we don't need to add number checking for `leftConst`